src/auth.ts

```diff
@@ -155,3 +156,48 @@ export async function getAuthorizationToken(
     throw Error(Errors.AUTH_TOKEN_URL_RESPONSE_DATA_MISSING);
   }
 }
+
+/**
+ * Refresh the access token in the given AuthorizationToken instance
+ *
+ * @param authToken auth token instance to be refreshed
+ * @param bb - instance of the SDK facade class
+ * @returns new auth token instance with refreshed access token
+ */
+export async function refreshAuthToken(
+  authToken: AuthorizationToken,
+  bb: BlueButton
+) {
+  const tokenUrl = getAccessTokenUrl(bb);
+  // const formData = new FormData();
+  // formData.append("username", bb.clientId);
+  // formData.append("passowrd", bb.clientSecret);
+  // formData.append("grant_type", "refresh_token");
+  // formData.append("client_id", bb.clientId);
+  // formData.append("refresh_token", authToken.refreshToken);
+  // const resp = await axios({
+  //     method: 'post',
+  //     url: tokenUrl,
+  //     data: formData,
+  //     headers: SDK_HEADERS,
+  // });
+
+  const resp = await axios.post(
+    tokenUrl,
+    {},
+    {
+      headers: SDK_HEADERS,
+      auth: {
+        username: bb.clientId,
+        password: bb.clientSecret,
+      },
+      params: {
+        grant_type: "refresh_token",
+        client_id: bb.clientId,
+        refresh_token: authToken.refreshToken,
+      },
```

Comment:
@JFU-GIT Just pointing out here, but this might be part of your BB2-1473 Jira story. Please ignore here, if to be refactored in that ticket.

This should be refactored to use the DATA part of the axios.post() request. 

`axios.post(url[, data[, config]])`

The following in an example of doing this with CURL:

```
curl -s -w "%{http_code}" -X POST "${BB_SERVER}/v1/o/token/" \
    -u "${BB_CLIENT_ID}:${BB_CLIENT_SECRET}" \
    -d "grant_type=refresh_token&client_id=${BB_CLIENT_ID}&refresh_token=${REFRESH_TOKEN}"

```

Reply:
thanks for comment, already tried removing 'params' and use 'data' instead on BB2-1473, still having issue unsupported grant type, still debugging on that work, probably keep it as a separate ticket for later issue and fix tracking.